llvm/lib/Transforms/IPO/AttributorAttributes.cpp

```diff
@@ -12491,6 +12491,9 @@ struct AAAddressSpaceImpl : public AAAddressSpace {
   void initialize(Attributor &A) override {
     assert(getAssociatedType()->isPtrOrPtrVectorTy() &&
            "Associated value is not a pointer");
+    auto *PtrTy = cast<PointerType>(getAssociatedType());
```

Comment:
I don't think you need this cast.

Reply:
well, it is needed as `getAddressSpace` is a method of `PointerType` instead of `Type`.